### PR TITLE
Magnification

### PR DESCRIPTION
**Pull request by roflmaostc**

### CODE REVIEW

**Comment by RainerHeintzmann** [comment]
Is it really necessary to have this magnification option? Also, if we have it, I suggest to not touch (or rather overwrite) the center settings. The user may well want a M=1 but specify her/his own center and it is surprising, if this does not work. If no center is specified, the code anyway picks the correct center.

**Comment by roflmaostc** [comment]
Ah, yes.

We can fix the latter one by choosing `dst_center` within the function signature.

**Comment by roflmaostc** [comment]
But, yes.

Maybe it is not needed.

File: src/selection_tools.jl [hunk 1]
```diff
@@ -377,7 +387,15 @@ julia> dst=select_region(a,new_size=(10,10), dst_center=(1,1)) # pad a with zero
...
 ```
 """
-function select_region(src::AbstractArray{T,N}; new_size=size(src), center=size(src).÷2 .+1, pad_value=zero(eltype(src)), dst_center = new_size .÷ 2 .+1) where {T,N}
+function select_region(src::AbstractArray{T,N}; M=nothing, new_size=size(src), center=size(src).÷2 .+1, pad_value=zero(eltype(src)), dst_center = new_size .÷ 2 .+1) where {T,N}
+
+    @assert isnothing(M) || new_size == size(src) "Either choose M or new_size, don't set both at the same time"
+
+    if !isnothing(M)
+        new_size = round.(Int, M .* size(src))
+        dst_center = new_size .÷ 2 .+ 1
```
Comment (RainerHeintzmann) [inline src/selection_tools.jl]: This line should be deleted.

Comment (roflmaostc) [reply src/selection_tools.jl]: fixed

**Comment by RainerHeintzmann** [comment]
Nice one. I like also the new assert line.

**Pull request merged**